enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,250 @@
+can cause the operator to not be installed properly. Allowing users to specify policy
+dependencies will alleviate these issues.
+
+### Goals
+
+1. Policy template dependencies: allow users to automatically process policy templates
+in a desired order and guarantee that this order will be stable
+2. Opt-in: any extra fields that are used to specify policy ordering should be optional,
+and policies that do not contain these fields should function as they did before in order
+to avoid issues with older policies
+
+### Non-Goals
+
+1. Status message matching: we plan to only allow users to conditionally process policy
+templates based on compliance state, so users will not be able to specify that they want a
+policy to be processed if another policy has a specific status message
+
+## Proposal
+
+We would add an optional new `dependencies` field to the spec on the configuration 
+policy level as well as an `extraDependencies` field on the template level to
+allow users to specify conditions for policy processing on both the policy and
+template level. When these fields are set and their conditions are not met, 
+the policy templates will not be applied on the managed cluster and the affected
+policy will enter a "pending" state. If this happens, any policy sets that contain
+the pending policy will also be marked as pending. When the conditions are met or
+the fields are not set, the policy templates will be processed normally.
+
+### User Stories
+
+1. As an author of a ConfigurationPolicy, I don’t want resources to be wasted when
+I know one object depends on another. If an object cannot be cleanly applied before
+another is created, I want to specify this in my Policy, and not waste
+resources (CPU, k8s API requests) in the policy controllers.
+2. As an author of a ConfigurationPolicy, I want to specify the order resources are
+applied in a policy when the API is not declarative. For some APIs (eg OLM), if
+resources are applied in the wrong order they can end up in an unhealthy state
+that I want to avoid.
+3. As a user of PolicySets, I want policies inside a set to be applied in a
+certain order.
+4. As an author of many ConfigurationPolicies, I have several policies that
+depend on another policy. I want to be able to reference completely separate
+policies (outside of the Policy “wrapper”) as required before a configuration
+is applied. (Maybe allow a policy to reference the status of a policy that runs
+before it (success or failure) before it tries to process it. Allowing you to chain
+them together)
+5. As a cluster administrator, I want to do a declarative action on a managed
+cluster when a policy becomes non-compliant. So I want to “activate” a
+ConfigurationPolicy based on the state of another policy in the cluster.
+6. As a user of the policy generator, I want to be able to generate policies that
+will have their templates be processed in a certain order.
+
```

Comment:
For mixed policy, how is the feature to handle ?  Two cases: 
1. Two or three types templates in a policy, for example, cert and configuration templates 
2. One cert type policy A is depend on another configuration type policy B.

Reply:
The handling of the dependencies will be done by the framework, not by the individual policy controllers. When a dependency is not met, the template will not be created on the managed cluster, and the framework will specially handle populating the status of that template in the Policy. So mixing types in a Policy is no problem, the framework will handle them all the same.

If a cert policy "A" depends on a config policy "B" to be compliant, then the framework will not create "A" on the managed cluster until "B" is compliant. And until then, the framework will mark "A" as `Pending`, and add a status message mentioning that it is waiting on "B".